sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOrcScan.scala

```diff
@@ -227,21 +270,25 @@ object GpuOrcScan extends Arm {
   def canCast(from: TypeDescription, to: TypeDescription): Boolean = {
     import org.apache.orc.TypeDescription.Category._
     if (!to.getCategory.isPrimitive || !from.getCategory.isPrimitive) {
+      // List, Map, Struct, Union is not primitive
       // Don't convert from any to complex, or from complex to any.
       // Align with what CPU does.
       return false
     }
+    val toType = to.getCategory
     from.getCategory match {
       case BOOLEAN | BYTE | SHORT | INT | LONG =>
-        to.getCategory match {
-          case BOOLEAN | BYTE | SHORT | INT | LONG => true
+        toType match {
+          case BOOLEAN | BYTE | SHORT | INT | LONG | FLOAT | DOUBLE | STRING |
+               TIMESTAMP => true
+          // BINARY and DATE are not supported by design.
+          // The 'to' type (aka read schema) is from Spark, and VARCHAR and CHAR will
+          // be replaced by STRING. Meanwhile, cuDF doesn't support them as output
+          // types, and also replaces them with STRING.
+          // TIMESTAMP_INSTANT is not supported by cuDF.
           case _ => false
         }
-      case VARCHAR =>
-        to.getCategory == STRING
-      // TODO more types, tracked in https://github.com/NVIDIA/spark-rapids/issues/5895
-      case _ =>
-        false
+      case _ => false
```

Comment:
This should not be removed. Here can see the raw types of ORC file. So `VARCHAR` and `CHAR` is visible. But they can not be seen in `castColumnTo` which is called after cudf reading, with  `VARCHAR` and `CHAR` being converted to `STRING`.